test/terra/noise/test_quantum_error.py

```diff
@@ -14,18 +14,224 @@
+            noise_1q.compose(noise_2q)
+
+    def test_compose_with_different_type_of_operator(self):
+        """Test compose with Kraus operator."""
+        noise_x = QuantumError([((IGate(), [0]), 0.9), ((XGate(), [0]), 0.1)])
+        meas_kraus = Kraus([np.diag([1, 0]),
+                            np.diag([0, 1])])
+        actual = noise_x.compose(meas_kraus)
+        expected = QuantumError([([(IGate(), [0]), (meas_kraus.to_instruction(), [0])], 0.9),
+                                 ([(XGate(), [0]), (meas_kraus.to_instruction(), [0])], 0.1)])
+        self.assertEqual(actual, expected)
+
+    def test_tensor(self):
+        """Test tensor two quantum errors."""
+        noise_x = QuantumError([((IGate(), [0]), 0.9), ((XGate(), [0]), 0.1)])
+        noise_y = QuantumError([((IGate(), [0]), 0.8), ((YGate(), [0]), 0.2)])
+        actual = noise_x.tensor(noise_y)
+        expected = QuantumError([([(IGate(), [1]), (IGate(), [0])], 0.9 * 0.8),
+                                 ([(IGate(), [1]), (YGate(), [0])], 0.9 * 0.2),
+                                 ([(XGate(), [1]), (IGate(), [0])], 0.1 * 0.8),
+                                 ([(XGate(), [1]), (YGate(), [0])], 0.1 * 0.2)])
+        self.assertEqual(actual, expected)
+
+    def test_expand(self):
+        """Test tensor two quantum errors."""
+        noise_x = QuantumError([((IGate(), [0]), 0.9), ((XGate(), [0]), 0.1)])
+        noise_y = QuantumError([((IGate(), [0]), 0.8), ((YGate(), [0]), 0.2)])
+        actual = noise_y.expand(noise_x)  # reversed order of expand
+        expected = QuantumError([([(IGate(), [1]), (IGate(), [0])], 0.9 * 0.8),
+                                 ([(IGate(), [1]), (YGate(), [0])], 0.9 * 0.2),
+                                 ([(XGate(), [1]), (IGate(), [0])], 0.1 * 0.8),
+                                 ([(XGate(), [1]), (YGate(), [0])], 0.1 * 0.2)])
+        self.assertEqual(actual, expected)
+
+    def test_tensor_with_different_type_of_operator(self):
+        """Test tensor with Kraus operator."""
+        noise_x = QuantumError([((IGate(), [0]), 0.9), ((XGate(), [0]), 0.1)])
+        meas_kraus = Kraus([np.diag([1, 0]),
+                            np.diag([0, 1])])
+        actual = noise_x.tensor(meas_kraus)
+        expected = QuantumError([([(IGate(), [1]), (meas_kraus.to_instruction(), [0])], 0.9),
+                                 ([(XGate(), [1]), (meas_kraus.to_instruction(), [0])], 0.1)])
+        self.assertEqual(actual, expected)
+
+
+# ================== Tests for old interfaces ================== #
+# TODO: remove after deprecation period
+import warnings
+class TestQuantumErrorOldInterface(common.QiskitAerTestCase):
+    """Testing the deprecating interface of QuantumError class"""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # overwrite the filter not to regard DeprecationWarning as error
+        warnings.filterwarnings("ignore", category=DeprecationWarning)
```

Comment:
We should probably target this down with the `module` argument as well (though it can be a little finnicky in some cases),